common/djangoapps/student/models.py

```diff
@@ -2982,3 +2982,21 @@ def retire_recovery_email(cls, user_id):
             pass
 
         return True
+
+
+class AllowedAuthEdxUser(TimeStampedModel):
+    EDX_EMAIL_DOMAIN = 'edx.org'
```

Comment:
I mentioned this in another PR, but please do not make code/models that are edX-specific. Many people outside of edX use the platform.

Reply:
If you need to make a place to store email domains that have to log in through some particular third party auth vendor, then please make it generic. So for example:

(Org:edX), [domains: edx.org, open.edx.org ...], [required_third_party_auth: google, ...], [exceptions: dave@edx.org, dormsbee@edx.org, ...]

Reply:
The above-suggested approach will work but it has too many related models. We can simplify this by adding a field `allowed_third_party_auth_domain` in the `SiteConfiguration` model which will have the domain name i.e `egx.org, edx.edge.org`. It will make this generic as you have suggested above.